From: Free Ekanayaka <freee@debian.org>
Date: Sat, 26 Nov 2016 10:40:11 +0000
Subject: sort-option-keys

Fix flaky twisted.test.test_main.MainTests.test_twisted which fails
if options are not in the same order.

See https://twistedmatrix.com/trac/attachment/ticket/8923.

---
 src/twisted/python/usage.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/twisted/python/usage.py b/src/twisted/python/usage.py
index 4a3a972..3335583 100644
--- a/src/twisted/python/usage.py
+++ b/src/twisted/python/usage.py
@@ -388,7 +388,7 @@ class Options(dict):
         dct = {}
         reflect.addMethodNamesToDict(self.__class__, dct, "opt_")
 
-        for name in dct.keys():
+        for name in sorted(dct.keys()):
             method = getattr(self, 'opt_'+name)
 
             takesArg = not flagFunction(method, name)
